### DISCUSSION TITLE
Google Analytics 4

### DISCUSSION THREAD

**Discussion by tresero** [razonyang/hugo-theme-bootstrap#773] [General]
Does this theme support GA4?

```
We'll soon sunset Universal Analytics
On July 1, 2023, Universal Analytics will stop processing new hits. If you still rely on Universal Analytics, we urge you to prepare for the future by moving over to Google Analytics 4, our next-generation measurement solution.

```

**Comment by razonyang** [comment]
Yes, the GA4 is supported since [v1.0.0-alpha.0](https://github.com/razonyang/hugo-theme-bootstrap/commit/2de8649f4317eda055c761a243fd5a4abaf4f4bc)

https://github.com/razonyang/hugo-theme-bootstrap/blob/b9c704cd7ec77c27cd86e5970a9def5d20e84df1/exampleSite/config/production/config.toml#L2

**Reply by tresero** [reply]
As always thanks!

**Comment by tresero** [comment]
BTW, you may want to add that in the comment on the config.

**Reply by razonyang** [reply]
Done, https://hbs.razonyang.com/v1/en/docs/configuration/site/, thanks for pointing it out.

**Reply by tresero** [reply]
No worries.

**Comment by tresero** [comment]
I'm not sure you are using the right GA4 javascript.

Check https://salsablanca.com it seems that it's using the old analytics.js?

I could be wrong of course, but it's not sending any events. I'm using the same site settings as the old site.

**Reply by razonyang** [reply]
Hi, does your `googleAnalytics`(config.toml) begin with `G-`? It should work since my blog site and the docs site are using GA4.

**Reply by razonyang** [reply]
Please also check the `config/_default/config.toml` and the `config/production/config.toml`, I guess that the `googleAnalytics` isn't correct, the GA4 ID should begin with `G-`.

**Comment by tresero** [comment]
I don't believe that's correct anymore.

https://support.google.com/analytics/answer/9304153#add-tag&zippy=%2Cadd-the-google-tag-directly-to-your-web-pages

**Comment by tresero** [comment]
I found it, their docs are so confusing.

**Comment by tresero** [comment]
Still doesn't work. I have no idea, is there anyway to just add this manually?

**Reply by razonyang** [reply]
I saw the GA4 script is shown correctly on your site now, just wait a minute to see if it works or not.
![image](https://user-images.githubusercontent.com/17720932/201969339-cec42c5d-568d-492e-a679-c918ba47dc15.png)

**Comment by tresero** [comment]
Yes, I guess it really isn't real time :)

Thanks again!

**Reply by razonyang** [reply]
You're welcome, please mark an answer if it helps.

**Reply by tresero** [reply]
How do I do that? All I see is smiley face

**Reply by razonyang** [reply]
@tresero Oops, sorry, I forgot that is under the `General` not `Q&A`, just ignore it.
